sql/plan/alter_table.go

```diff
@@ -53,6 +53,127 @@ func (r *RenameTable) String() string {
 	return fmt.Sprintf("Rename table %s to %s", r.OldNames, r.NewNames)
 }
 
+func (r *RenameTable) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	// TODO: 'ALTER TABLE newViewName RENAME view1' should fail on renaming views - should be fixed in vitess
+	renamer, ok := r.Db.(sql.TableRenamer)
+	if !ok {
+		return nil, sql.ErrRenameTableNotSupported.New(r.Db.Name())
+	}
+	viewDb, vok := r.Db.(sql.ViewDatabase)
+
+	for i, oldName := range r.OldNames {
+		if success, err := r.attemptRenameTable(ctx, oldName, r.NewNames[i]); err != nil {
+			return nil, err
+		} else if success {
+			err = renamer.RenameTable(ctx, oldName, r.NewNames[i])
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			if success, err = r.attemptRenameView(ctx, viewDb, vok, oldName, r.NewNames[i]); err != nil {
+				return nil, err
+			} else if !success {
+				return nil, sql.ErrTableNotFound.New(oldName)
+			}
+		}
+	}
```

Comment:
This is a little harder to read than it could be...  try first checking if the table exists, and if so, then call `renameTable`, then if it doesn't exist, check if the view exists and call `renameView` and then if it doesn't exist, return the ErrTableNotFound error. 

If you use that structure, this logic will be a little easier to read and it'll probably help you simplify the interfaces for the two rename functions, too.